concordium-sdk/src/main/java/com/concordium/sdk/ClientV2MapperExtensions.java

```diff
@@ -100,4 +128,332 @@ static BlockIdentifier to(FinalizedBlockInfo finalizedBlockInfo) {
                 .blockHeight(to(finalizedBlockInfo.getHeight()))
                 .build();
     }
+
+    static com.concordium.grpc.v2.AccountIdentifierInput to(AccountRequest accountIdentifier) {
+        var builder = AccountIdentifierInput.newBuilder();
+
+        switch (accountIdentifier.getType()) {
+            case ADDRESS:
+                builder.setAddress(to(accountIdentifier.getAddress()));
+                break;
+            case INDEX:
+                builder.setAccountIndex(to(accountIdentifier.getIndex()));
+                break;
+            case CREDENTIAL_REGISTRATION_ID:
+                builder.setCredId(to(accountIdentifier.getRegistrationId()));
+                break;
+        }
+
+        return builder.build();
+    }
+
+    static CredentialRegistrationId to(com.concordium.sdk.transactions.CredentialRegistrationId registrationId) {
+        return CredentialRegistrationId.newBuilder()
+                .setValue(to(registrationId.getRegId()))
+                .build();
+    }
+
+    static AccountIndex to(com.concordium.sdk.responses.AccountIndex index) {
+        return AccountIndex.newBuilder().setValue(index.getIndex().getValue()).build();
+    }
+
+    static AccountAddress to(com.concordium.sdk.transactions.AccountAddress address) {
+        return AccountAddress.newBuilder()
+                .setValue(to(address.getBytes()))
+                .build();
+    }
+
+    static com.concordium.sdk.responses.accountinfo.AccountInfo to(AccountInfo account) {
+        var builder = com.concordium.sdk.responses.accountinfo.AccountInfo.builder()
+                .accountNonce(to(account.getSequenceNumber()))
+                .accountAmount(to(account.getAmount()))
+                .accountReleaseSchedule(to(account.getSchedule()))
+                .accountCredentials(ImmutableMap.copyOf(to(
+                        account.getCredsMap(),
+                        k -> Index.from(k),
+                        ClientV2MapperExtensions::to)))
+                .accountThreshold(account.getThreshold().getValue())
+                .accountEncryptedAmount(to(account.getEncryptedBalance()))
+                .accountEncryptionKey(ElgamalPublicKey.from(account.getEncryptionKey().getValue().toByteArray()))
+                .accountIndex(to(account.getIndex()))
+                .accountAddress(to(account.getAddress()));
+
+        if (account.hasStake()) {
+            if (account.getStake().hasBaker()) {
+                builder.bakerInfo(to(account.getStake().getBaker()));
+            } else if (account.getStake().hasDelegator()) {
+                builder.delegation(to(account.getStake().getDelegator()));
+            } else {
```

Comment:
You don't need need case I think. At least it should have a proper error message. 

But if you look at the `.proto` definition then it's a `oneof` 
```
oneof staking_info {
    // The account is a baker.
    Baker baker = 1;
    // The account is a delegator.
    Delegator delegator = 2;
  }
```

So I think a simple `if..else` is appropriate, because if the account is staking then it's either a baker or a delegator.